imap_processing/ena_maps/utils/naming.py

```diff
@@ -23,8 +26,350 @@ class MappableInstrumentShortName(Enum):
+        if instrument in [
+            MappableInstrumentShortName.LO,
+            MappableInstrumentShortName.LO_HI_RES,
+            MappableInstrumentShortName.LO_HI_THROUGHPUT,
+        ]:
+            sensor = f"{sensor:03}"
+        else:
+            raise ValueError("Integer sensor values are only valid for LO instruments.")
+    # Hi and Ultra may be either "45", "90", or "combined", in which case
+    # Hi should get the sensor "ic" and Ultra should get the sensor "lc"
+    elif sensor == "combined":
+        if instrument is MappableInstrumentShortName.ULTRA:
+            sensor = "lc"
+        elif instrument is MappableInstrumentShortName.HI:
+            sensor = "ic"
+
+    # Get the instrument descriptor (e.g. "u90", "h45", "ilo")
+    instrument_descriptor = f"{instrument.value}{sensor}"
+    return instrument_descriptor
+
+
+def parse_instrument_descriptor(
+    instrument_descriptor: str,
+) -> tuple[MappableInstrumentShortName, str]:
+    """
+    Parse the instrument descriptor string into instrument, sensor str reprs.
+
+    Parameters
+    ----------
+    instrument_descriptor : str
+        The instrument descriptor string to parse.
+
+    Returns
+    -------
+    tuple[str, str]
+        A tuple containing the instrument short name and the sensor.
+
+    Raises
+    ------
+    ValueError
+        If the instrument descriptor format can't be parsed as a combined instrument or
+        as a regex match to the expected format.
+    """
+    # Default to no sensor
+    sensor = ""
+
+    if instrument_descriptor.endswith("c"):
+        sensor = "combined"
+        instrument_short_name = instrument_descriptor[:-2]
+        instrument = MappableInstrumentShortName(instrument_short_name)
+    else:
+        # Do regex to get the instrument and sensor:
+        # The first 1 or 3 characters are the instrument short name
+        # if the last 2 or 3 characters are digits, then this is a sensor
+        # otherwise, the sensor is empty str
+        match = re.match(r"([a-z]{1,3})(\d{2,3})?", instrument_descriptor)
+        if match:
+            instrument_short_name = match.group(1)
+            sensor = match.group(2) if match.group(2) else ""
+            instrument = MappableInstrumentShortName(instrument_short_name)
+    instrument = MappableInstrumentShortName(instrument_short_name)
+    return instrument, sensor
+
+
+def parse_map_duration(
+    duration: str | int | timedelta,
+) -> str:
+    """
+    Parse the duration into a string representation.
+
+    Parameters
+    ----------
+    duration : str | int | timedelta
+        The duration to parse. This can be a string in the format "1yr", "6mo", etc.,
+        an integer representing the number of days, or a timedelta object.
+
+    Returns
+    -------
+    str
+        The parsed duration string in the format "1yr", "6mo", etc.
+    """
+    if isinstance(duration, timedelta):
+        # Convert timedelta to a string representation of number of 28.5 day months
```

Comment:
Where does 28.5 days come from? Is that something from IBEX? We should document why we are choosing that number somewhere here because it is non-obvious to me.

Reply:
Tim and I were just discussing this in [another PR](https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/1756#discussion_r2098254613).

Short answer: I vaguely recall finding an edge case where 28.5 day months was a better simple assumption than 30 day months; however I can't replicate it, so I'll switch to 30 day months here.